### PR TITLE
fix Dockerfile to use look aside cache

### PR DESCRIPTION
**Pull request by jcantrill**
This PR modifies the Dockerfile to utilize the look aside cache.  When dist-git repos available,  we will probably need to re-add the sources and regen the source file:

`rhpkg new-sources <dependences>`

Validated it builds downstream in the existing repos with:

https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=15697170

### CODE REVIEW

File: elasticsearch/Dockerfile [hunk 1]
```diff
@@ -1,70 +1,80 @@
...
+RUN while read -r hash file ; do \
+      curl -s -o $file http://pkgs.devel.redhat.com/repo/rpms/logging-elasticsearch-docker/$file/$hash/$file ; \
+    done <sources
 
-RUN yum-config-manager --enable rhel-7-server-ose-3.7-rpms \
+RUN if [ -n "${LOCAL_REPO}" ] ; then \
+     curl -s -o /etc/yum.repos.d/local.repo ${LOCAL_REPO} ; \
+    fi
+
+RUN yum-config-manager --enable rhel-7-server-ose-3.8-rpms \
```
Comment (richm) [inline elasticsearch/Dockerfile]: 3.9 or maybe 3.10?

File: elasticsearch/prep-install.prod [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # prep the plugins we need
+pushd /tmp/lib
+
 es_plugins=()
-for es_plugin in openshift-elasticsearch-plugin elasticsearch-cloud-kubernetes
+for es_plugin in $(ls *.zip)
```
Comment (richm) [inline elasticsearch/prep-install.prod]: `for es_plugin in *.zip`

File: elasticsearch/Dockerfile [hunk 3]
```diff
@@ -1,70 +1,80 @@
...
 USER 1000
 CMD ["sh", "/opt/app-root/src/run.sh"]
+
+LABEL \
+        io.k8s.description="Elasticsearch container for EFK aggregated logging storage" \
+        com.redhat.component="logging-elasticsearch-docker" \
+        vendor="Red Hat" \
+        name="openshift3/logging-elasticsearch" \
+        License="GPLv2+" \
+        io.k8s.display-name="Elasticsearch 2.4.4" \
```
[the remark this answers is not in the document]
Comment (jcantrill) [reply elasticsearch/Dockerfile]: fixed

File: elasticsearch/prep-install.prod (hunk 2 above)
Comment (jcantrill) [reply elasticsearch/prep-install.prod]: fixed

File: elasticsearch/Dockerfile (hunk 1 above)
Comment (jcantrill) [reply elasticsearch/Dockerfile]: from my conversations with others familiar with brew, this line is either modified directly or ignored completely so going to leave as is

**Pull request merged**